Tests/FoundationEssentialsTests/FileManager/FileManagerTests.swift

```diff
@@ -1046,6 +1046,67 @@ private struct FileManagerTests {
         }
     }
 
+    #if !os(Windows) && !os(WASI) && !os(OpenBSD) && !canImport(Android)
+    @Test func extendedAttributesDoNotFollowSymlinksWhenSetting() async throws {
+        let xattrKey = FileAttributeKey("NSFileExtendedAttributes")
+        #if canImport(Darwin)
+        let attrName = "com.swiftfoundation.symlinktest"
+        #elseif os(Linux)
+        // Linux requires the user.* namespace prefix for regular files
+        let attrName = "user.swiftfoundation.symlinktest"
+        #else
+        let attrName = "swiftfoundation.symlinktest"
+        #endif
+        let attrValue = Data([0xAA, 0xBB, 0xCC])
+
+        try await FilePlayground {
+            File("target", contents: Data("payload".utf8))
+            SymbolicLink("link", destination: "target")
+        }.test { fileManager in
+            // Attempt to set xattrs on the symlink; if unsupported, bail quietly.
+            do {
+                try fileManager.setAttributes(
+                    [xattrKey: [attrName: attrValue]], ofItemAtPath: "link")
+            } catch let error as CocoaError {
+                if error.code == .featureUnsupported {
+                    return
+                }
+                if let posix = error.underlying as? POSIXError,
+                    posix.code == .ENOTSUP || posix.code == .EOPNOTSUPP
+                {
+                    return
+                }
+                throw error
+            }
+
+            let linkAttrs = try fileManager.attributesOfItem(atPath: "link")
+            let targetAttrs = try fileManager.attributesOfItem(atPath: "target")
+
+            let linkXattrs = linkAttrs[xattrKey] as? [String: Data]
+            let targetXattrs = targetAttrs[xattrKey] as? [String: Data]
+
+            try #require(
+                linkXattrs != nil || targetXattrs != nil,
+                "Expected extended attributes after successful setAttributes call")
```

Comment:
Typically we use `#require` to unwrap values:

```suggestion
            let linkXattrs = try #require(linkAttrs[xattrKey] as? [String: Data])
            let targetXattrs = try #require(targetAttrs[xattrKey] as? [String: Data])
```

That way you can remove the `if` check below because the `else` branch will never run, but using `#require` to unwrap values enforces this at compile time instead of runtime.